src/js/api/core-shim.js

```diff
@@ -65,6 +65,15 @@ const CoreShim = function(originalContainer) {
     ], () => true);
 };
 
+if (__HEADLESS__) {
+    CoreShim.prototype.set = function(property, value) {
```

Comment:
The method is added from the start to avoid TypeErrors from early use, but also to allow values to be set right after calling setup. There's no more risk in that than using set inappropriately at any point. The SDK team has to use this at their own discretion. It is not intended for public use. I added a table of attributes that are appropriate to set in the headless player for the SDK team to use as a guide https://jwplayer.atlassian.net/wiki/spaces/JW8/pages/1016660021/Headless+JavaScript+Player#6.-Private-model-access

There may be others that make sense in very specific cases (like the [chromecast integration](https://jwplayer.atlassian.net/wiki/spaces/JW8/pages/1016660021/Headless+JavaScript+Player#ChromeCast-Reference) but this is not a document covering the player model and how to hack it for fun and profit).